web/crux/src/app/image/image.http.controller.ts

```diff
@@ -1,43 +1,106 @@
-import { Controller, Post, Body, UseGuards, UseInterceptors, Patch, UseFilters } from '@nestjs/common'
-import { ApiBody, ApiCreatedResponse, ApiOkResponse } from '@nestjs/swagger'
-import { AuditLogLevel } from 'src/decorators/audit-logger.decorator'
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  Param,
+  Patch,
+  Post,
+  Put,
+  UseFilters,
+  UseGuards,
+  UseInterceptors,
+} from '@nestjs/common'
+import { ApiBody, ApiCreatedResponse, ApiNoContentResponse, ApiOkResponse, ApiTags } from '@nestjs/swagger'
+import { Identity } from '@ory/kratos-client'
 import HttpExceptionFilter from 'src/filters/http-exception.filter'
-import { Empty } from 'src/grpc/protobuf/proto/common'
-import { AddImagesToVersionRequest, ImageListResponse, PatchImageRequest } from 'src/grpc/protobuf/proto/crux'
 import HttpLoggerInterceptor from 'src/interceptors/http.logger.interceptor'
 import PrismaErrorInterceptor from 'src/interceptors/prisma-error-interceptor'
-import { AddImagesToVersionRequestDto, ImageListResponseDto, PatchImageRequestDto } from 'src/swagger/crux.dto'
-import { Identity } from '@ory/kratos-client'
+import { CreatedResponse, CreatedWithLocation } from '../shared/created-with-location.decorator'
+import CreatedWithLocationInterceptor from '../shared/created-with-location.interceptor'
 import JwtAuthGuard, { IdentityFromRequest } from '../token/jwt-auth.guard'
+import ImageAddToVersionTeamAccessGuard from './guards/image.add-to-version.team-access.guard'
+import ImageOrderImagesTeamAccessGuard from './guards/image.order-images.team-access.guard'
+import ImageTeamAccessGuard from './guards/image.team-access.guard'
+import { AddImagesDto, ImageDto, PatchImageDto } from './image.dto'
 import ImageService from './image.service'
-import ImagePatchValidationPipe from './pipes/image.patch.pipe'
+import ImageAddToVersionValidationInterceptor from './interceptors/image.add-images.interceptor'
+import DeleteImageValidationInterceptor from './interceptors/image.delete.interceptor'
+import OrderImagesValidationInterceptor from './interceptors/image.order.interceptor'
+
+const ProductId = () => Param('productId')
+const VersionId = () => Param('versionId')
+const ImageId = () => Param('imageId')
+
+const ROUTE_IMAGE_ID = ':imageId'
 
-@Controller('image')
-@UseGuards(JwtAuthGuard)
-@UseInterceptors(HttpLoggerInterceptor, PrismaErrorInterceptor)
+@Controller('/products/:productId/versions/:versionId/images')
+@ApiTags('version-images')
+@UseGuards(JwtAuthGuard, ImageTeamAccessGuard)
+@UseInterceptors(HttpLoggerInterceptor, PrismaErrorInterceptor, CreatedWithLocationInterceptor)
 @UseFilters(HttpExceptionFilter)
 export default class ImageHttpController {
   constructor(private service: ImageService) {}
 
+  @Get()
+  @ApiOkResponse({ type: Array<ImageDto> })
```

Comment:
Swagger can't parse these `Array` types. Proper solution: 
```js@ApiOkResponse({ type: ImageDto, isArray:true })``` please refactor it.